codalab/model/bundle_model.py

```diff
@@ -2910,18 +2961,71 @@ def create_bundle_store(self, user, name, storage_type, storage_format, url, aut
             connection.execute(cl_bundle_store.insert().values(bundle_store_value))
         return uuid
 
-    def save_oauth2_auth_code(self, grant):
-        with self.engine.begin() as connection:
-            result = connection.execute(oauth2_auth_code.insert().values(grant.columns))
-            grant.id = result.lastrowid
-        return grant
-
-    def delete_oauth2_auth_code(self, auth_code_id):
+    def update_bundle_store(self, user, uuid, update_fields):
+        """
+        update a bundle store
+        """
         with self.engine.begin() as connection:
             connection.execute(
-                oauth2_auth_code.delete().where(oauth2_auth_code.c.id == auth_code_id)
+                cl_bundle_store.update()
+                .where(and_(cl_bundle_store.c.uuid == uuid, cl_bundle_store.c.owner_id == user,))
+                .values(update_fields)
             )
 
+    def get_bundle_store(self, user, uuid):
+        """
+        Return the bundle store corresponding to the specified uuid.
+        Just returns the owner_id, name, and url of the bundle store.
+        """
+        with self.engine.begin() as connection:
+            row = connection.execute(
+                select(
+                    [
+                        cl_bundle_store.c.uuid,
+                        cl_bundle_store.c.owner_id,
+                        cl_bundle_store.c.name,
+                        cl_bundle_store.c.storage_type,
+                        cl_bundle_store.c.storage_format,
+                        cl_bundle_store.c.url,
+                    ]
+                ).where(
+                    and_(
+                        cl_bundle_store.c.uuid == uuid,
+                        or_(
+                            cl_bundle_store.c.owner_id == 'codalab(0)',
```

Comment:
same issue with root user as mentioned above

Reply:
sorry, whats the issue?

Reply:
I said in the other comment:

```
We can't assume the root user is called "codalab" or has id "0". Also, isn't owner_id supposed to be an int and not a string "codalab(0)"?

You should use the root user id configured from here:

codalab-worksheets/codalab/lib/codalab_manager.py

Line 403 in be5d92d

 def root_user_id(self): 
Also I'd test this out to double check it actually gets the root user's bundle.

 
Member
@epicfaace epicfaace 22 hours ago
Just use self.root_user_id which is set to the root user's id.
```